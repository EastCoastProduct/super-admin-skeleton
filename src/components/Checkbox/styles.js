import { StyleSheet } from 'aphrodite/no-important';
import { colors, sizes } from '../../styles/variables';

export default StyleSheet.create({
  base: {
    display: 'none',
  },
  checkbox: {
    backgroundColor: colors.header,
    borderRadius: sizes.radius,
    color: colors.input,
    cursor: 'pointer',
    display: 'inline-block',
    height: 25,
    marginRight: 10,
    textAlign: 'center',
    verticalAlign: 'middle',
    width: 25,
  },
  checked: {
    fontSize: 25,
  },
});
