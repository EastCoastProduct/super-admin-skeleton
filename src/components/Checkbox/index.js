import React, { PropTypes } from 'react';
import { css } from 'aphrodite/no-important';
import styles from './styles';

function Checkbox(props) {
  const { className, id, label, input, meta: _, onChange, ...rest } = props;

  return (
    <label
      className={css(className && className)}
      htmlFor={`${id}-${input.name}`}
    >
      <span className={css(styles.checkbox, input.value && styles.checked)}>
        {input.value && <i className="fa fa-check" />}
      </span>
      <input
        className={css(styles.base)}
        {...input}
        {...rest}
        id={`${id}-${input.name}`}
        type="checkbox"
        onChange={(e) => {
          input.onChange(e);
          if (onChange) setTimeout(() => onChange());
        }}
      />
      {label}
    </label>
  );
}

Checkbox.propTypes = {
  className: PropTypes.object,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
  onChange: PropTypes.func,
};

export default Checkbox;
