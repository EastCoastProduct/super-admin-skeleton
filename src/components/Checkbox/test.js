import React from 'react';
import { shallow } from 'enzyme';
import deepMerge from 'deepmerge';
import { StyleSheet } from 'aphrodite/no-important';
import Checkbox from './';

describe('Checkbox component', () => {
  const props = {
    id: 'UsersFilters',
    label: 'This is checkbox',
    input: {
      name: 'checkbox',
      value: false,
      onChange: jest.fn(),
    },
    meta: {},
    onChange: jest.fn(),
  };
  const wrapper = shallow(<Checkbox {...props} />);

  describe('snapshot', () => {
    it('renders basic required data', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders with custom onChange event and checked', () => {
      const newProps = deepMerge(props, {
        input: {
          value: true,
        },
      });
      const newWrapper = shallow(<Checkbox {...newProps} />);

      expect(newWrapper).toMatchSnapshot();
    });

    it('renders with additional styles and attributes', () => {
      const styles = StyleSheet.create({
        additional: {
          margin: 0,
          padding: 0,
        },
      });
      const newProps = deepMerge(props, {
        className: styles.additional,
        disabled: true,
        type: 'text',
      });
      const wrapper = shallow(<Checkbox {...newProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('instance', () => {
    const instance = wrapper.instance();

    it('trigger onChange event', () => {
      const e = { target: { checked: true } };
      wrapper.find('input').simulate('change', e);
      jest.runAllTimers();

      expect(instance.props.input.onChange).toHaveBeenCalledWith(e);
      expect(instance.props.onChange).toHaveBeenCalled();
    });
  });
});
