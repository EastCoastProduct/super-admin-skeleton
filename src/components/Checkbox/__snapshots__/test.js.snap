exports[`Checkbox component snapshot renders basic required data 1`] = `
<label
  className=""
  htmlFor="UsersFilters-checkbox">
  <span
    className="checkbox_1gmevj5" />
  <input
    className="base_avg348"
    id="UsersFilters-checkbox"
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value={false} />
  This is checkbox
</label>
`;

exports[`Checkbox component snapshot renders with additional styles and attributes 1`] = `
<label
  className="additional_1433df3"
  htmlFor="UsersFilters-checkbox">
  <span
    className="checkbox_1gmevj5" />
  <input
    className="base_avg348"
    disabled={true}
    id="UsersFilters-checkbox"
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value={false} />
  This is checkbox
</label>
`;

exports[`Checkbox component snapshot renders with custom onChange event and checked 1`] = `
<label
  className=""
  htmlFor="UsersFilters-checkbox">
  <span
    className="checkbox_1gmevj5-o_O-checked_1ehyjl1">
    <i
      className="fa fa-check" />
  </span>
  <input
    className="base_avg348"
    id="UsersFilters-checkbox"
    name="checkbox"
    onChange={[Function]}
    type="checkbox"
    value={true} />
  This is checkbox
</label>
`;
